### PR TITLE
chore(fp-ts): refactor aspect ratio helpers

### PR DESCRIPTION
**Pull request by frederickfogerty**
This is part of the Gatsby fp-ts refactor.

### CODE REVIEW

**Comment by frederickfogerty** [comment]
Current dependencies on/for this PR:
* main
  * **PR #198** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/198" target="_blank">Graphite</a> 
    * **PR #188** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/188" target="_blank">Graphite</a> 
      * **PR #189** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/189" target="_blank">Graphite</a> 
        * **PR #191** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/191" target="_blank">Graphite</a> 
          * **PR #192** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/192" target="_blank">Graphite</a> 
            * **PR #193** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/193" target="_blank">Graphite</a> 
              * **PR #194** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/194" target="_blank">Graphite</a> 
                * **PR #195** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/195" target="_blank">Graphite</a> 
                  * **PR #196** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/196" target="_blank">Graphite</a> 
                    * **PR #197** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/197" target="_blank">Graphite</a> 
                      * **PR #199** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/199" target="_blank">Graphite</a> 
                        * **PR #201** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/201" target="_blank">Graphite</a>  (this PR)
                          * **PR #202** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/202" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/imgix/gatsby/201?utm_source=stack-comment).

File: src/common/ar.ts [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-  pipe(
-    (v: StringAspectRatio) => v.split(':') as [string, string],
-    map((part) => parseFloat(part)),
-    ([width, height]) => width / height,
-    (v) => roundToDP(3, v),
-  )(ar);
+  const splitAR = validatedAR.value.split(':') as [string, string];
+  const [parsedWidth, parsedHeight] = splitAR.map((part) => parseFloat(part));
+  if (Number.isNaN(parsedWidth) || Number.isNaN(parsedHeight)) {
+    throw new Error('AR is not valid');
```
Comment (luqven) [inline src/common/ar.ts:21]: nit: same as the comment above, would be nice to add the reason AR is not valid here.

**Review by luqven — APPROVED** [review]
LGTM 👍🏼

**Review by sherwinski — APPROVED** [review]

File: src/common/ar.ts [hunk 2]
```diff
@@ -1,31 +1,25 @@
...
- * Validates that an aspect ratio is in the format w:h. If false is returned, the aspect ratio is in the wrong format.
+ * Parse a string AR into a float.
  */
-function aspectRatioIsValid(aspectRatio: string): boolean {
-  if (typeof aspectRatio !== 'string') {
-    return false;
+export const parseStringARParam = (ar: StringAspectRatio): number => {
+  const validatedAR = StringAspectRatioSchema.validate(ar);
+  if (validatedAR.error) {
+    throw new Error('AR is not valid');
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply src/common/ar.ts:16]: Good call-out, improved this in a follow up PR

File: src/common/ar.ts (hunk 1 above)
Comment (frederickfogerty) [reply src/common/ar.ts:21]: Good call-out, improved this in a follow up PR

**Pull request merged**